src/libhirte/common/cfg.c

```diff
@@ -137,6 +141,29 @@ int cfg_load_from_file(struct config *config, const char *config_file) {
         return ini_parse(config_file, parsing_handler, config);
 }
 
+char *is_file_name_ending_with_conf(char *file_name, char *conf_file_enfing) {
+        return strstr(file_name, conf_file_enfing);
+}
+
+int cfg_load_from_dir(struct config *config, const char *custom_config_directory) {
+        struct dirent **all_files_in_dir = NULL;
```

Comment:
I haven't run it in valgrind yet, but I think you have to free `all_files_in_dir` as `scandir` calls `malloc` to allocate the memory for the items.

Reply:
Added free function for all_files_in_dir veriable